src/Compilers/CSharp/Portable/Symbols/Source/TypeParameterConstraintClause.cs

```diff
@@ -111,55 +114,40 @@ internal sealed class TypeParameterConstraintClause
             }
 
             Debug.Assert((constraints & TypeParameterConstraintKind.ObliviousNullabilityIfReferenceType) == 0 ||
-                         (constraints & ~(TypeParameterConstraintKind.ObliviousNullabilityIfReferenceType | TypeParameterConstraintKind.Constructor | TypeParameterConstraintKind.Default)) == 0);
+                         (constraints & ~(TypeParameterConstraintKind.ObliviousNullabilityIfReferenceType | TypeParameterConstraintKind.Constructor | TypeParameterConstraintKind.Default |
+                                          TypeParameterConstraintKind.PartialMismatch | TypeParameterConstraintKind.ValueTypeFromConstraintTypes | TypeParameterConstraintKind.ReferenceTypeFromConstraintTypes)) == 0);
 #endif 
             this.Constraints = constraints;
             this.ConstraintTypes = constraintTypes;
-            this.IgnoresNullableContext = ignoresNullableContext;
         }
 
         public readonly TypeParameterConstraintKind Constraints;
         public readonly ImmutableArray<TypeWithAnnotations> ConstraintTypes;
-        public readonly bool IgnoresNullableContext;
 
         internal bool IsEmpty => Constraints == TypeParameterConstraintKind.None && ConstraintTypes.IsEmpty;
 
-        /// <summary>
-        /// Adjust unresolved instances of <see cref="TypeWithAnnotations"/> which represent nullable type parameters
-        /// of the <paramref name="container"/> within constraint types according to the IsValueType state inferred
-        /// from the constraint types.
-        /// </summary>
-        internal static void AdjustConstraintTypes(Symbol container, ImmutableArray<TypeParameterSymbol> typeParameters,
-                                                   ArrayBuilder<TypeParameterConstraintClause> constraintClauses,
-                                                   ref IReadOnlyDictionary<TypeParameterSymbol, bool> isValueTypeOverride)
+        internal static Dictionary<TypeParameterSymbol, bool> BuildIsValueTypeMap(Symbol container, ImmutableArray<TypeParameterSymbol> typeParameters,
```

Comment:
>BuildIsValueTypeMap [](start = 62, length = 19)

nit: consider moving this method over to ConstraintsHelper since that the only place it's used. Same for `BuildIsReferenceTypeFromConstraintTypesMap` below.